backend/core/query_logger_dual.py

```
@@ -0,0 +1,490 @@
+"""
+Enhanced query logging service that writes to both JSON and SQLite.
+
+This module extends the original query logger to write to both:
+1. JSON files (for backward compatibility and data export)
+2. SQLite database (for real-time admin dashboard access)
+"""
+
+import hashlib
```

Ruff (0.12.2)

`hashlib` imported but unused

Remove unused import: `hashlib`

(F401)